src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,244 @@
+     * @throws InvalidTaskIndexException When the index is out of bounds.
+     * @throws InvalidFormatException    When the input command is of the wrong format.
+     */
+    public String done(String command) throws InvalidTaskIndexException, InvalidFormatException {
+        int taskIdx = Parser.getIndexFromCommand(command);
+
+        // Handle invalid index
+        if (taskIdx >= 1 && taskIdx <= tasks.size()) {
+            Task t = tasks.get(taskIdx - 1);
+            t.markAsDone();
+            return String.format("Nice! I've marked this task as done:\n  %s\n", t);
+        } else {
+            throw new InvalidTaskIndexException();
+        }
+    }
+
+    /**
+     * Adds a new todo task to the TaskList.
+     *
+     * @param command The input command from the user.
+     * @return Formatted string containing the added todo task, to be displayed to the user.
+     * @throws InvalidFormatException        When the input command is of the wrong format.
+     * @throws EmptyTodoDescriptionException When the description is empty.
+     */
+    public String todo(String command) throws InvalidFormatException, EmptyTodoDescriptionException {
+        String[] matches = Parser.validateRegexAndMatch(command, "^todo (.+)", "todo {description}");
+        String description = matches[1].trim();
+        if (description.equals("")) {
+            throw new EmptyTodoDescriptionException();
+        }
+        Todo newTodo = new Todo(description);
+        this.tasks.add(newTodo);
+        return this.formatAddTaskString(newTodo);
+    }
+
+    /**
+     * Adds a new deadline task to the TaskList.
+     *
+     * @param command The input command from the user.
+     * @return Formatted string containing the added deadline task, to be displayed to the user.
+     * @throws InvalidFormatException When the input command is of the wrong format.
+     */
+    public String deadline(String command) throws InvalidFormatException {
+        String[] matches = Parser.validateRegexAndMatch(command, "^deadline (.+) /by (.+)",
+                "deadline {description} /by {date}");
+
+        LocalDate date;
+        try {
+            date = LocalDate.parse(matches[2].trim());
+        } catch (DateTimeParseException e) {
+            throw new InvalidFormatException("Please ensure the date is in YYYY-MM-DD format!");
+        }
+
+        Deadline newDeadline = new Deadline(matches[1].trim(), date);
+        tasks.add(newDeadline);
+        return formatAddTaskString(newDeadline);
+    }
+
+    /**
+     * Adds a new event task to the TaskList.
+     *
+     * @param command The input command from the user.
+     * @return Formatted string containing the added event task, to be displayed to the user.
+     * @throws InvalidFormatException When the input command is of the wrong format.
+     */
+    public String event(String command) throws InvalidFormatException {
```

Comment:
Perhaps a name that is a verb would be better here? To align with the coding conventions of names presenting methods being verbs.